macos/FluentUI/Separator.swift

```diff
@@ -0,0 +1,82 @@
+//
+//  Copyright (c) Microsoft Corporation. All rights reserved.
+//  Licensed under the MIT License.
+//
+
+import AppKit
+
+@objc(MSFSeparatorStyle)
+public enum SeparatorStyle: Int {
+    case `default`
+    case shadow
+
+    fileprivate var color: NSColor {
+		switch self {
+		case .default:
+			if #available(OSX 10.14, *) {
+				return NSColor.separatorColor
+
+			} else {
+				return NSColor.systemGray
+			}
+		case .shadow:
+			return NSColor.gridColor
+        }
+    }
+}
+
+@objc(MSFSeparatorOrientation)
+public enum SeparatorOrientation: Int {
+    case horizontal
+    case vertical
+}
+
+
+@objc(MSFSeparator)
+open class Separator: NSView {
+	private var orientation: SeparatorOrientation = .horizontal
+	
+	private var style: SeparatorStyle = .default
```

Comment:
Since it's private this is not mutable? Or am I missing something? If it's not mutable, we can make it a let and we also don't need the updateLayer/wantsUpdateLayer. We can just set it once on init and overall do less work when the system asks us to display

Reply:
I think right now these things are `var`s because they're populated in `initialize` which gets called _after_ `super.init(...)`. I think if we were to set them before, (which might result in minor code duplication) we could make them `let`s

Reply:
Didn't know you could have a let without assigning it till the init. Thanks

Reply:
@anandrajeswaran I'm fairly sure I still need `updateLayer/wantsUpdateLayer` so that the background color changes with light/dark mode. With testing, I could get away with not overriding `wantsUpdateLayer`, but I'm pretty sure that's just undefined behavior.